examples/circom.rs

```diff
@@ -0,0 +1,119 @@
+use std::fmt::Debug;
+use std::marker::PhantomData;
+use std::sync::Arc;
+use std::time::Instant;
+
+use lurk::circuit::gadgets::circom::CircomGadget;
+use lurk::circuit::gadgets::pointer::AllocatedPtr;
+use lurk::coprocessor::circom::non_wasm::CircomCoprocessor;
+use lurk::eval::{empty_sym_env, lang::Lang};
+use lurk::field::LurkField;
+use lurk::proof::{nova::NovaProver, Prover};
+use lurk::ptr::Ptr;
+use lurk::public_parameters::public_params;
+use lurk::store::Store;
+use lurk::{Num, Symbol};
+use lurk_macros::Coproc;
+use pasta_curves::pallas::Scalar as Fr;
+
+const REDUCTION_COUNT: usize = 1;
+
+#[derive(Debug, Clone)]
+pub struct CircomSha256<F: LurkField> {
+    _n: usize,
+    pub(crate) _p: PhantomData<F>,
+}
+
+impl<F: LurkField> CircomSha256<F> {
+    fn new(n: usize) -> Self {
+        CircomSha256 {
+            _n: n,
+            _p: PhantomData,
+        }
+    }
+}
+
+impl<F: LurkField> CircomGadget<F> for CircomSha256<F> {
+    fn name(&self) -> &str {
+        "circom_sha256"
+    }
+
+    fn into_circom_input(self, _input: &[AllocatedPtr<F>]) -> Vec<(String, Vec<F>)> {
+        // TODO: actually use the lurk inputs
+        let arg_in = ("arg_in".into(), vec![F::ZERO, F::ZERO]);
+        vec![arg_in]
+    }
+
+    fn simple_evaluate(&self, s: &mut Store<F>, _args: &[Ptr<F>]) -> Ptr<F> {
+        // TODO: actually use the lurk inputs
+        let expected = Num::Scalar(
+            F::from_str_vartime(
+                "55165702627807990590530466439275329993482327026534454077267643456",
+            )
+            .unwrap(),
+        );
+        s.intern_num(expected)
+    }
+}
+
+#[derive(Clone, Debug, Coproc)]
+enum Sha256Coproc<F: LurkField> {
+    SC(CircomCoprocessor<F, CircomSha256<F>>),
+}
+
+/// Run the example in this file with
+/// `cargo run --release --example circom`
```

Comment:
Except I don't have a circom source at hand as part of this PR, do I? It sure would be nice to have the module-level documentation tell me what pre-requisite commands I need to issue for this example to work.

Reply:
Done